File: src/cli/user/me.ts
import type { Command } from "commander";
import { getClient } from "../../lib/client.ts";
import { printError, printJson } from "../../lib/output.ts";

export function registerMe(user: Command): void {
  user
    .command("me")
    .description("Current authenticated user")
    .action(async () => {
      try {
        const client = getClient();
        const viewer = await client.viewer;
        const org = await viewer.organization;
        printJson({
          id: viewer.id,
          name: viewer.name,
          email: viewer.email,
          displayName: viewer.displayName,
          organization: { id: org.id, name: org.name },
        });
      } catch (err) {
        printError(err instanceof Error ? err.message : "Failed to get user");
      }
    });
}

File: internal/cli/user/me.go
package user

import (
	"context"

	"github.com/spf13/cobra"

	"github.com/shhac/lin/internal/linear"
	"github.com/shhac/lin/internal/output"
)

func registerMe(user *cobra.Command) {
	cmd := &cobra.Command{
		Use:   "me",
		Short: "Current authenticated user",
		Args:  cobra.NoArgs,
		Run: func(_ *cobra.Command, _ []string) {
			client := linear.GetClient()

			resp, err := linear.Viewer(context.Background(), client)
			if err != nil {
				output.HandleGraphQLError(err)
			}

			v := resp.Viewer
			output.PrintJSON(map[string]any{
				"id":          v.Id,
				"name":        v.Name,
				"email":       v.Email,
				"displayName": v.DisplayName,
				"organization": map[string]any{
					"id":   v.Organization.Id,
					"name": v.Organization.Name,
				},
			})
		},
	}
	user.AddCommand(cmd)
}
